### PR TITLE
January 2022 newsletter draft

### PR DESCRIPTION
**Pull request by milancurcic**
Sorry for the last-minute submission. Considering people may be offline around New Year's eve, let's plan to complete and merge this by Wednesday, January 5, if possible

* [x] PRs summary
* [x] GitHub stats data
* [x] Flang updates by @AlexisPerry 
* [x] LFortran updates by @certik

### CODE REVIEW

File: _posts/2022-01-01-Fortran-Newsletter-January-2022.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+  --env switch lets you specify the prefix of the compiler-related environment variables
+* [#539](https://github.com/fortran-lang/fpm/pull/539) (WIP):
+  Add parent packages into dependency tree
+* [#498](https://github.com/fortran-lang/fpm/pull/498) (WIP):
+  Compiler flags profiles
+
+`fpm` is still in early development and we need as much help as we can get.
+Here's how you can help today:
+
+* Use it and let us know what you think! Read the [fpm packaging guide](https://github.com/fortran-lang/fpm/blob/main/PACKAGING.md)
```
Comment (awvwgk) [inline _posts/2022-01-01-Fortran-Newsletter-January-2022.md:159]:
```suggestion
* Use it and let us know what you think! Read the [fpm packaging guide](https://fpm.fortran-lang.org/en/tutorial)
```

File: _posts/2022-01-01-Fortran-Newsletter-January-2022.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+to learn how to build your package with fpm, and the [manifest reference](https://github.com/fortran-lang/fpm/blob/main/manifest-reference.md)
```
Comment (awvwgk) [inline _posts/2022-01-01-Fortran-Newsletter-January-2022.md:160]:
```suggestion
to learn how to build your package with fpm, and the [manifest reference](https://fpm.fortran-lang.org/en/spec/manifest.html)
```

File: _posts/2022-01-01-Fortran-Newsletter-January-2022.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+## Contributors
+
+We thank everybody who contributed to fortran-lang in the past month by
+commenting in any of these repositories:
+
+* [fortran-lang/stdlib](https://github.com/fortran-lang/stdlib)
+* [fortran-lang/stdlib-cmake-example](https://github.com/fortran-lang/stdlib-cmake-example)
+* [fortran-lang/fpm](https://github.com/fortran-lang/fpm)
+* [fortran-lang/fpm-registry](https://github.com/fortran-lang/fpm-registry)
```
Comment (awvwgk) [inline _posts/2022-01-01-Fortran-Newsletter-January-2022.md:204]:
```suggestion
* [fortran-lang/fpm-registry](https://github.com/fortran-lang/fpm-registry)
* [fortran-lang/fpm-docs](https://github.com/fortran-lang/fpm-docs)
```

**Review by awvwgk — COMMENTED** [review]
Let's update the links to the new fpm pages

File: _posts/2022-01-01-Fortran-Newsletter-January-2022.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+  Add MCST LCC C, C++ and Fortran compiler
+* [#351](https://github.com/fortran-lang/fortran-lang.org/pull/351):
+  Use HEAD to reference default branch
+* [#355](https://github.com/fortran-lang/fortran-lang.org/pull/355):
+  2021 review article draft
+
+### Work in progress
+
+* [#356](https://github.com/fortran-lang/fortran-lang.org/pull/356) (WIP):
+  Adding Fortran Error Handler to packages index
```
Comment (jvdp1) [inline _posts/2022-01-01-Fortran-Newsletter-January-2022.md:33]: I believe that it has been merged.
```suggestion
```

File: _posts/2022-01-01-Fortran-Newsletter-January-2022.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+  Newsletter draft for December 2021
+* [#350](https://github.com/fortran-lang/fortran-lang.org/pull/350):
+  Updated CaNS item so that it shows the version
+* [#353](https://github.com/fortran-lang/fortran-lang.org/pull/353):
+  Add MCST LCC C, C++ and Fortran compiler
+* [#351](https://github.com/fortran-lang/fortran-lang.org/pull/351):
+  Use HEAD to reference default branch
+* [#355](https://github.com/fortran-lang/fortran-lang.org/pull/355):
+  2021 review article draft
+
```
Comment (jvdp1) [inline _posts/2022-01-01-Fortran-Newsletter-January-2022.md:29]:
```suggestion
* [#356](https://github.com/fortran-lang/fortran-lang.org/pull/356) (WIP):
  Adding Fortran Error Handler to packages index

```

File: _posts/2022-01-01-Fortran-Newsletter-January-2022.md [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+### LFortran
+
+TODO @certik
+
+We are looking for new contributors. Please do not hesitate to contact us if
+you are interested. We will help you get up to speed.
+
+## Events
+
```
Comment (milancurcic) [inline _posts/2022-01-01-Fortran-Newsletter-January-2022.md:186]: I forgot about fpm-docs. I think we should also mention it under Events.

```suggestion

* fpm has a new documentation website hosted at
  [fpm.fortran-lang.org](https://fpm.fortran-lang.org/).
  This website will provide user-oriented tutorials and how-to guides, as well
  as developer-oriented reference documents and specifications.
  We welcome all contributions to the fpm documentation, including translations
  to other languages.
  Please visit the [fpm-docs repo](https://github.com/fortran-lang/fpm-docs) to
  get started.
```

File: _posts/2022-01-01-Fortran-Newsletter-January-2022.md [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+* [fortran-lang/fpm](https://github.com/fortran-lang/fpm)
+* [fortran-lang/fpm-registry](https://github.com/fortran-lang/fpm-registry)
+* [fortran-lang/fpm-docs](https://github.com/fortran-lang/fpm-docs)
+* [fortran-lang/setup-fpm](https://github.com/fortran-lang/setup-fpm)
+* [fortran-lang/fpm-haskell](https://github.com/fortran-lang/fpm-haskell)
+* [fortran-lang/fortran-lang.org](https://github.com/fortran-lang/fortran-lang.org)
+* [fortran-lang/benchmarks](https://github.com/fortran-lang/benchmarks)
+* [fortran-lang/fortran-forum-article-template](https://github.com/fortran-lang/fortran-forum-article-template)
+* [fortran-lang/fftpack](https://github.com/fortran-lang/fftpack)
+* [j3-fortran/fortran\_proposals](https://github.com/j3-fortran/fortran_proposals)
```
Comment (jvdp1) [inline _posts/2022-01-01-Fortran-Newsletter-January-2022.md:212]: Should `testdrive` be in this list?

**Review by jvdp1 — COMMENTED** [review]
Thank you @milancurcic for starting the newsletter

File: _posts/2022-01-01-Fortran-Newsletter-January-2022.md [hunk 8]
```diff
@@ -0,0 +1,214 @@
+---
+layout: post
+title: "Fortran newsletter: January 2022"
+category: newsletter
+author: Milan Curcic
```
Comment (milancurcic) [inline _posts/2022-01-01-Fortran-Newsletter-January-2022.md:5]:
```suggestion
author: Milan Curcic, Sebastian Ehlert, Jérémie Vandenplas
```

File: _posts/2022-01-01-Fortran-Newsletter-January-2022.md [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+
+* [#634](https://github.com/fortran-lang/fpm/pull/634):
+  Better extraction of the Fortran compiler from the MPI wrapper
+
+### Work in progress
+
+* [#642](https://github.com/fortran-lang/fpm/pull/642) (WIP):
+  Replace polymorphic assignment with move_alloc
+* [#630](https://github.com/fortran-lang/fpm/pull/630) (WIP):
+  just allow . on new subcommand instead of changing canonical path pro…
```
Comment (awvwgk) [inline _posts/2022-01-01-Fortran-Newsletter-January-2022.md:146]:
```suggestion
  allow backfilling of current directory in new subcommand
```

File: _posts/2022-01-01-Fortran-Newsletter-January-2022.md (hunk 7 above)
Comment (awvwgk) [reply _posts/2022-01-01-Fortran-Newsletter-January-2022.md:212]:
```suggestion
* [fortran-lang/test-drive)[https://github.com/fortran-lang/test-drive)
* [j3-fortran/fortran\_proposals](https://github.com/j3-fortran/fortran_proposals)
```

**Comment by AlexisPerry** [comment]
Flang updates are available here: https://github.com/milancurcic/fortran-lang.github.io/pull/3

**Comment by milancurcic** [comment]
I'm confused about something here. [This PR](https://github.com/milancurcic/fortran-lang.github.io/pull/3) by @AlexisPerry introduced an unintended commit ( https://github.com/milancurcic/fortran-lang.github.io/pull/3/commits/c6509ffa9c33f3070265a7697aeb19f5377d0b32), which I missed because only the newsletter file was listed under GitHub's Files changed tab. But now that that PR is merged, I see that the changes to _data/package_index.yml are included here, but they shouldn't be.

@AlexisPerry @awvwgk do you understand why this is and how to fix it? I'm pretty sure this has to do with my not understanding git to a sufficient level.

**Comment by AlexisPerry** [comment]
That's my fault, sorry!  I saw changes to that file showing up in my PR and undid them since I only wanted to update the newsletter file.  I don't know how to remove specific commits, but if someone does, this is the one that need removing:
- d2924471d925cc04614a528358d3e1629bc47ec7

**Comment by milancurcic** [comment]
Okay, it seems like it's this commit https://github.com/milancurcic/fortran-lang.github.io/pull/3/commits/d2924471d925cc04614a528358d3e1629bc47ec7. Is there an easy way to revert only that one commit?

**Comment by jvdp1** [comment]
Could you try `git revert d292447`?

Le mar. 4 janv. 2022 à 22:37, Milan Curcic ***@***.***> a
écrit :

**Comment by milancurcic** [comment]
@jvdp1 Wonderful, that seems to have worked. :)

File: _posts/2022-01-01-Fortran-Newsletter-January-2022.md [hunk 10]
```diff
@@ -0,0 +1,313 @@
...
+* TargetRewrite: Rewrite fir.address_of(func)
+* Fix folding of EXPONENT() intrinsic function
+* OPEN(RECL=) handling for sequential formatted I/O
+* Avoid potential deadlock in CloseAll()
+
+Call notes are recorded and available upon request [here](https://docs.google.com/document/d/10T-S2J3GrahpG4Ooif93NSTz2zBW0MQc_RlwHi0-afY). Please contact Alexis Perry-Holby at aperry@lanl.gov for document access.
+
+### LFortran
+
+TODO @certik
```
Comment (certik) [inline _posts/2022-01-01-Fortran-Newsletter-January-2022.md:270]:
```suggestion
* Beginning of refactoring of ASR (Abstract Semantic Representation) into a standalone library
* New intrinsics: mvbits, bge, bgt, ble, blt, ibits
* generic procedure resolution fixes
* FreeBSD fixes
* Implement `.xor.` (LFortran as well as GFortran extension)
* Fixes to large integers (BigInt) handling
* Added support for `private`, `final` attributes in derived types/classes

The following people contributed code in December 2021: Ondřej Čertík, Gagandeep Singh, Harris Snyder
```

File: _posts/2022-01-01-Fortran-Newsletter-January-2022.md [hunk 11]
```diff
@@ -0,0 +1,313 @@
+---
+layout: post
+title: "Fortran newsletter: January 2022"
+category: newsletter
+author: Milan Curcic, Sebastian Ehlert, Jérémie Vandenplas, Alexis Perry-Holby
```
Comment (certik) [inline _posts/2022-01-01-Fortran-Newsletter-January-2022.md:5]:
```suggestion
author: Milan Curcic, Sebastian Ehlert, Jérémie Vandenplas, Alexis Perry-Holby, Ondřej Čertík
```

**Comment by milancurcic** [comment]
Thank you all for your contributions! Let's publish tomorrow as planned.

**Comment by certik** [comment]
Thanks Milan for pushing it!

**Pull request merged**